### ISSUE TITLE
Bump up shared secret limit to 512?

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [FreeRADIUS/freeradius-client#119]
In practice, I doubt anyone uses shared secrets of more than 256 characters. However, a [Google search](https://github.com/radcli/radcli/issues/103#issuecomment-2100693335) provided some empirical information about maximal lengths for various implementations. The [AWS Directory Service](https://aws.amazon.com/fr/directoryservice/) is documented to have an exceptional upper limit of 512.

From the [AWS Directory Service Documentation](https://docs.aws.amazon.com/directoryservice/latest/devguide/API_RadiusSettings.html):
> Length Constraints: Minimum length of 8. Maximum length of 512.

**Comment by alandekok** [comment]
We can update the length to 512, but I doubt many people will use it.  There really is little point to any shared secret longer than 256 characters.  If people are that worried about security, they should use TLS.

**Comment by DimitriPapadopoulos** [comment]
I totally agree, even 96 was overkill. It's just that some might set a long secret just because it's possible.

For example, see discussion in _[Please enlarge the maximum Shared secret lengt in RADIUS Authenticattion Server settings](https://community.ui.com/questions/Please-enlarge-the-maximum-Shared-secret-lengt-in-RADIUS-Authenticattion-Server-settings/06bcf334-2cfe-4985-8cb4-dfc8433c42ae#answer/40caf845-3b86-447b-ae75-a4e0c45991e7)_:
> Thanks for your reply, but it still doesn't change the fact that my client is allready running this on multiple locations from different suppliers, and only Unifi doesn't support this while Microsoft RADIUS server does go up to 128 chars. I can't change the source. Can only ask Ubiquity to follow Microsoft standards.
